solr/core/src/java/org/apache/solr/util/ThreadCpuTimer.java

```diff
@@ -87,23 +87,27 @@ public long getStartCpuTimeNs() {
    *
    * @return current value, or {@link #UNSUPPORTED} if not supported.
    */
-  public long getCurrentCpuTimeNs() {
+  public long getElapsedTimerCpuNs() {
     if (THREAD_MX_BEAN != null) {
       return this.startCpuTimeNanos != UNSUPPORTED
-          ? THREAD_MX_BEAN.getCurrentThreadCpuTime() - this.startCpuTimeNanos
+          ? getThreadTotalCpuNs() - this.startCpuTimeNanos
           : UNSUPPORTED;
     } else {
       return UNSUPPORTED;
     }
   }
 
+  public long getThreadTotalCpuNs() {
+    return THREAD_MX_BEAN.getCurrentThreadCpuTime();
```

Comment:
Since it's a public method we need to check for null here, as in other methods.

Also, to be consistent with other names it should be named something like `getTotalCpuTimeNs()` ... although at this point the names become confusing again. Maybe we should use `elapsed` and `absolute` instead?

Reply:
Ah I spent a bunch of time trying to come up with names and didn't think to use the word absolute. That might work, Ill fiddle with that.